### PR TITLE
Add adopters for Karmada

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add adopters page for Karmada

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: adopters/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+[Here](adopters.md) lists the organisations who adopted the Karmada project in production.
+
+You just need to add an entry for your company and upon merging it will automatically be added to our website.
+
+To add your organisation follow these steps:
+
+1. Fork the [karmada-io/website](https://github.com/karmada-io/website) repository.
+2. Clone it locally with `git clone https://github.com/<YOUR-GH-USERNAME>/website.git`.
+3. (Optional) Add the logo of your organisation to `static/img/supporters`. Good practice is for the logo to be called e.g. `<company>.png`. This will only be used on our website.
```
Comment (RainbowMango) [inline adopters/README.md:15]: Example PR: https://github.com/karmada-io/website/pull/55

File: docs/casestudies/adopters.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+To join this list, please follow [these instructions](https://github.com/karmada-io/website/tree/main/adopters/README.md).
+
+## Karmada Adopters
+
+All organisations are sorted alphabetically.
+
+* [ci123](ci123/ci123.md)
+
+* [vipkid](vipkid/vipkid.md)
```
Comment (RainbowMango) [inline docs/casestudies/adopters.md:17]: table

File: adopters/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Karmada Adopters
+
+So you and your organisation are using Karmada? That's great. We would love to hear from you! 💖
+
+## Adding yourself
+
+[Here](adopters.md) lists the organisations who adopted the Karmada project in production.
```
Comment (RainbowMango) [inline adopters/README.md:7]: invalid link.

Comment (Poor12) [reply adopters/README.md]: Done.

**Comment by RainbowMango** [comment]
This branch has conflicts that must be resolved

**Comment by Poor12** [comment]
> This branch has conflicts that must be resolved

Done.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**